pkg/controllers/dynakube/activegate/internal/statefulset/builder/modifiers/eec.go

```diff
@@ -0,0 +1,81 @@
+package modifiers
+
+import (
+	dynatracev1beta2 "github.com/Dynatrace/dynatrace-operator/pkg/api/v1beta2/dynakube"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/activegate/consts"
+	"github.com/Dynatrace/dynatrace-operator/pkg/controllers/dynakube/activegate/internal/statefulset/builder"
+	"github.com/Dynatrace/dynatrace-operator/pkg/util/kubeobjects/container"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+)
+
+var _ volumeModifier = EecModifier{}
+var _ volumeMountModifier = EecModifier{}
+var _ builder.Modifier = EecModifier{}
+
+const (
+	eecVolumeName = "eec-token"
+	eecDir        = "/var/lib/dynatrace/secrets/eec"
+	eecFile       = "eec.token"
+)
+
+func NewEecVolumeModifier(dynakube dynatracev1beta2.DynaKube) EecModifier {
+	return EecModifier{
+		dynakube: dynakube,
+	}
+}
+
+type EecModifier struct {
+	dynakube dynatracev1beta2.DynaKube
+}
+
+func (mod EecModifier) Enabled() bool {
+	return true
+}
+
+func (mod EecModifier) Modify(sts *appsv1.StatefulSet) error {
+	baseContainer := container.FindContainerInPodSpec(&sts.Spec.Template.Spec, consts.ActiveGateContainerName)
+	sts.Spec.Template.Spec.Volumes = append(sts.Spec.Template.Spec.Volumes, mod.getVolumes()...)
+	baseContainer.VolumeMounts = append(baseContainer.VolumeMounts, mod.getVolumeMounts()...)
+
+	baseContainer.Command = []string{"/bin/sh"}
+	baseContainer.Args = []string{
+		"-c",
+		"echo abc ; cp -v " + eecDir + "/" + eecFile + " /var/lib/dynatrace/gateway/config/ ; /opt/dynatrace/gateway/entrypoint.sh",
+	}
```

Comment:
The `cp` command is not needed any more. AG-ExtensionController can read from '/var/lib/dynatrace/secrets/eec/token` directory (AG v 1.296.0.20240628)

Reply:
@aorcholski If you already changed this, please close the discussions to make the PR more readable